uvm/impl/com/untangle/uvm/AppManagerImpl.java

```diff
@@ -1136,10 +1147,42 @@ public void doAutoInstall()
         float totalMemoryMb = ((float) ((double) totalMemory / 1024)) / (1024.0f);
         int totalTnterfaceCount = UvmContextFactory.context().networkManager().getNetworkSettings().getInterfaces().size();
 
-        for (AppProperties appProps : getAllAppProperties()) {
-            if (!appProps.getAutoInstall()) continue;
-            if(appProps.getAutoInstallMinMemory() > totalMemoryMb) continue;
-            if(appProps.getAutoInstallMinRequireInterfaces() > totalTnterfaceCount) continue;
+        LicenseManager lm = UvmContextFactory.context().licenseManager();
+
+        List<AppProperties> appPropsToInstall = getAllAppProperties();
+
+        if (lm.isRestricted()) {
+            appPropsToInstall = this.restrictedAllowedApps;
+        }
+
+        for (AppProperties appProps : appPropsToInstall) {
+            if (!lm.isRestricted() && !appProps.getAutoInstall()) {
+                logger.info("App not able to be autoinstalled: " + appProps.getName());
+                continue;
+            }
+            if(appProps.getAutoInstallMinMemory() > totalMemoryMb) {
+                logger.info("Not enough memory to install app " + appProps.getName());
+                continue;
+            }
+            if(appProps.getAutoInstallMinRequireInterfaces() > totalTnterfaceCount) {
+                logger.info("Not enough interfaces to install app: " + appProps.getName());
+                continue;
+            }
+            if (appInstances(appProps.getName()).size() >= 1) {
+                logger.info("App already installed: " + appProps.getName());
+                // start app 
+                for (App app: appInstances(appProps.getName())) {
+                    if (app.getAppProperties().getAutoStart()) {
+                        try {
```

Comment:
only start apps that are not installed and not running - installed apps that are not running should not be auto started